### ISSUE TITLE
What should I look for (2.3.2 broke my compile)

### ISSUE THREAD

**Issue by iherbivore** [dlmanning/gulp-sass#510]
I am using a variation of the mgechev angular2-seed with sass. 
I have a sass option property:

    SASS_OPTIONS: SassOptions = {
    includePaths: [
      './node_modules/bootstrap-sass/assets/stylesheets',
      './node_modules/font-awesome/scss',
      './src/css'
    ]};

and a task that looks like this:

    function processComponentScss() {
    return gulp.src([
      join(APP_SRC, '**', '*.scss'),
      '!' + join(APP_SRC, 'assets', '**', '*.scss')
    ])
    .pipe(plugins.sass(SASS_OPTIONS).on('error', plugins.sass.logError))
    .pipe(isProd ? plugins.cached('process-component-scss') : plugins.util.noop())
    .pipe(plugins.postcss(processors))
    .pipe(gulp.dest(isProd ? TMP_DIR: APP_DEST));
    }

Everything works with 2.3.1 but with 2.3.2 it can't find an import - `@import "myvariables";`

I have the scss file:  `./src/css/_myvariables.scss`

I don't really understand what the 2.3.2 change was saying. Can you give me a general direction here?
Thanks.

**Comment by xzyfer** [comment]
Please remove everything that isn't gulp-sass to confirm it's not another plugin doing something wrong.

The change in 2.3.2 prevent the settings object you pass into gulp-sass being changed. In your case the `SASS_OPTIONS` variable would be altered by gulp-sass for every file.

**Comment by iherbivore** [comment]
Sadly, that would be all but impossible. The advanced version of the angular2-seed is plugins plugins plugins. Looks like 2.3.1 is the one to keep.

**Comment by xzyfer** [comment]
I don't understand. Just comment out the stuff before and after sass in `processComponentScss()`.

```js
function processComponentScss() {
  return gulp.src([
    join(APP_SRC, '**', '*.scss'),
    '!' + join(APP_SRC, 'assets', '**', '*.scss')
  ])
  .pipe(plugins.sass(SASS_OPTIONS).on('error', plugins.sass.logError))
  .pipe(gulp.dest(TMP_DIR));
}
```

**Comment by iherbivore** [comment]
Oh I see. I didn't understand what you were saying. I will try that today.

**Comment by iherbivore** [comment]
OK, I understand, now. I added an 'on('end'...' to see what was happening to the SASS_OPTIONS.

    function processComponentScss() {
    return gulp.src([
        join(APP_SRC, '**', '*.scss'),
        '!' + join(APP_SRC, 'assets', '**', '*.scss')
    ])
        .pipe(plugins.sass(SASS_OPTIONS).on('error', plugins.sass.logError))
        .on('end', () => {
            console.log('there will be no more data.');
            console.log(SASS_OPTIONS);
        })
        .pipe(isProd ? plugins.cached('process-component-scss') : plugins.util.noop())
        .pipe(plugins.postcss(processors))
        .pipe(gulp.dest(isProd ? TMP_DIR : APP_DEST));
    }

Each path that contained scss files was added to the initial `includePaths` array in my `SASS_OPTIONS` which I guess is what you were saying. Seeing this, it quickly became apparent that my entry of `./src/css` should have been `./src/client/css`. It happened to be the first path processed by sass so it was always accidentally included for the files in the other paths. 
It was a simple mistake/correction, and I had almost given up. 
Thanks for your help!

**Comment by xzyfer** [comment]
> Each path that contained scss files was added to the initial includePaths array in my SASS_OPTIONS

Are you saying `SASS_OPTIONS.includePaths` was being changed by gulp-sass? 
Was this happening on 2.3.1 or 2.3.2?

**Comment by iherbivore** [comment]
With both versions I start with 
    
    includePaths: [
       './node_modules/bootstrap-sass/assets/stylesheets',
       './node_modules/font-awesome/scss',
       './src/css'
     ]

The log out put I added after the sass instruction with 2.3.2 looks the same, no changes. With 2.3.1 it looks more like this

    includePaths: [
      '/home/foo/project/src/client/app/login',
      '/home/foo/project/src/client/app',
      '/home/foo/project/src/client/css',
      './node_modules/bootstrap-sass/assets/stylesheets',
      './node_modules/font-awesome/scss',
      './src/css'
    ]

There are more paths than that, but the gist is that each folder processed by sass was getting prepended to the `includePaths`. I did not make any changes between the 2 runs except for the change between 2.3.1 and 2.3.2

**Comment by xzyfer** [comment]
I see, thank you. This was the intention of the fix that was in 2.3.2. Sorry it broke your build.
